### PR TITLE
initial support for CodeQL

### PR DESCRIPTION
**Pull request by pwntester**

### CODE REVIEW

File: queries/ql/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  "then"
+  "where"
+
+  (false)
+  (predicate)
+  (result)
+  (specialId)
+  (super)
+  (this)
+  (true)
```
Comment (theHamsta) [inline queries/ql/highlights.scm:45]: Thanks for the contribution. A few notes on the different group of keywords:
true, false -> `@boolean`
this, super -> `@variable.builtin`
sum,rank,instanceof,avg... -> `@function.builtin`
import -> `@include`
if, then... -> `@conditional`
and, or,select... -> `@keyword.operator`

File: queries/ql/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  ","
+  "|"
+] @punctuation.delimiter
+
+(className) @type
+
+(varName) @variable
+
+(integer) @number
+(float) @number
```
Comment (theHamsta) [inline queries/ql/highlights.scm:92]: `@float`

**Review by theHamsta — APPROVED** [review]

**Comment by theHamsta** [comment]
Strange the Github CI didn't add the parsers to the README...

**Comment by theHamsta** [comment]
Ah, it does not update README because you're on your master branch. We will get the README update on the next PR automatically that comes from another branch.

**Comment by vigoux** [comment]
Hmm that really needs a fix...

**Comment by theHamsta** [comment]
I mentioned "master" as a exception in the workflow. lsp-config workflow has the same problem.

**Comment by pwntester** [comment]
Thanks guys! will use a new branch in the next PR (locals.scm) so the README is generated

**Comment by theHamsta** [comment]
I already pushed that autogenerated commit

**Pull request merged**